nilearn/surface/surface.py

```diff
@@ -1975,3 +1976,243 @@ def new_img_like(ref_img, data):
         mesh=copy.deepcopy(mesh),
         data=data,
     )
+
+
+def _compute_adjacency_matrix(mesh, values="ones", dtype=None):
+    """Compute the adjacency matrix for a surface.
+
+    The adjacency matrix is a matrix
+    with one row and one column for each vertex
+    such that the value of a cell `(u,v)` in the matrix is 1
+    if nodes `u` and `v` are adjacent and 0 otherwise.
+
+    Parameters
+    ----------
+    mesh : InMemoryMesh
+
+    values : { 'len' | 'invlen' | 'ones'}, optional
+        If `values` is `'ones'` (the default), then the returned matrix
+        contains uniform values in the cells representing edges.
+        If the value is `'len'` then the cells contain
+        the edge length of the represented edge.
+        If the value is `'invlen'`, then the the inverse of the distances
+        are returned.
+
+    dtype : numpy dtype-like or None, optional
+        The dtype that should be used for the returned sparse matrix.
+
+    Returns
+    -------
+    matrix : scipy.sparse.csr_matrix
+        A sparse matrix representing the edge relationships in `surface`.
+
+    """
+    # This is a bit of a hack to quickly find a unique set of all edges.
+    if values not in {"len", "invlen", "ones"}:
+        raise ValueError(f"unrecognized values argument: {values}")
```

Comment:
given we control what value is passed to this function I am not sure it makes sense to control the inputs here